app/assets/javascripts/audits/batch-allocation.js

```diff
@@ -0,0 +1,38 @@
+window.GOVUKAdmin = window.GOVUKAdmin || {};
+window.GOVUKAdmin.Modules = window.GOVUKAdmin.Modules || {};
+
+(function (Modules) {
+  "use strict";
+
+  Modules.BatchAllocation = function () {
+    this.start = function ($root) {
+      this._initElements($root);
+      this._bindEventHandlers();
+    };
+
+    this._initElements = function ($root) {
+      this.$batchSizeInput = $root.find("#batch_size");
+      this.$checkboxes = $root.find("input[type=checkbox]");
+    };
+
+    this._bindEventHandlers = function () {
+      this.$batchSizeInput.keyup($.proxy(this.onBatchSizeChange, this));
```

Comment:
The fact that `this` is used three times on this line is a code smell to me. Any reason we don't just wrap this entire thing in `this.start`, and declare internal variables at the top of that function? Then we can stop saying `this` so much (which - in my experience - is generally a good thing).

Reply:
The module is a [constructor object][1], which means that it returns an object overtime it is instantiated. It is also a Singleton, as per the `Module` definition, with a public interface, by contract: the method `start`.

So, it is awkward to me to define everything int the start method, and not used the `this` object to define functions and instance variables. 

I am addressing your changes so we get this feature out, and ask in the frontend channel for best practices around Modules within GOV.UK.

[1]: https://github.com/alphagov/govuk_frontend_toolkit/blob/master/javascripts/govuk/modules.js#L35-L37

Reply:
Yes, happy to have discussion around this. My experience is that the JavaScript `this` keyword usually leads to more pain than it's worth, and should be avoided where possible. 

I'm happy to not define everything inside `this.start`; if you want to do that, then I'd prefer avoiding instance variables. Instead we should explicitly passing them around in as arguments, where `this.start` is like a "Main" method, which orchestrates all variable assignment and function calls.